Code/Core/FileIO/FileIO.cpp

```diff
@@ -212,17 +267,39 @@
     struct stat stat_source;
     VERIFY( lstat( srcFileName, &stat_source ) == 0 );
 
-    // Special case symlinks.
-    if ( S_ISLNK( stat_source.st_mode ) )
+    if ( linkBehavior == FOLLOW_NEITHER )
     {
-        AString linkPath( stat_source.st_size + 1 );
-        ssize_t length = readlink( srcFileName, linkPath.Get(), linkPath.GetReserved() );
-        if ( length != stat_source.st_size )
+        // Special case symlinks.
+        if ( S_ISLNK( stat_source.st_mode ) )
         {
+            AString linkPath( stat_source.st_size + 1 );
+            ssize_t length = readlink( srcFileName, linkPath.Get(), linkPath.GetReserved() );
+            if ( length != stat_source.st_size )
+            {
+                return false;
+            }
+            linkPath.SetLength( length );
+            if ( symlink( linkPath.Get(), dstFileName ) == 0 )
+            {
+                return true;
+            }
+            // symlink will not clobber.
+            if ( errno == EEXIST )
```

Comment:
I think this EEXIST test is probably a needed fix even without the rest of this PR